File: mall-nest-service/apps/system/src/address.controller.ts
import { Controller, Get, Post, Put, Delete, Body, Param, Query, ParseIntPipe } from '@nestjs/common';
import { AddressService } from './address.service';

@Controller('address')
export class AddressController {
  constructor(private addressService: AddressService) {}

  /**
   * 获取地址列表
   */
  @Get('list')
  async getAddressList(@Query('customerId', ParseIntPipe) customerId: number) {
    return this.addressService.getAddressList(customerId);
  }

  /**
   * 获取地址详情
   */
  @Get(':id')
  async getAddressDetail(@Param('id', ParseIntPipe) id: number) {
    return this.addressService.getAddressDetail(id);
  }

  /**
   * 新增地址
   */
  @Post('add')
  async addAddress(@Body() addressData: any) {
    return this.addressService.addAddress(addressData);
  }

  /**
   * 更新地址
   */
  @Put('update')
  async updateAddress(@Body() addressData: any) {
    return this.addressService.updateAddress(addressData);
  }

  /**
   * 删除地址
   */
  @Delete(':id')
  async deleteAddress(
    @Param('id', ParseIntPipe) id: number,
    @Query('customerId', ParseIntPipe) customerId: number,
  ) {
    return this.addressService.deleteAddress(id, customerId);
  }

  /**
   * 设置默认地址
   */
  @Put('default/:id')
  async setDefault(
    @Param('id', ParseIntPipe) id: number,
    @Query('customerId', ParseIntPipe) customerId: number,
  ) {
    return this.addressService.setDefault(id, customerId);
  }

  /**
   * 获取默认地址
   */
  @Get('default')
  async getDefaultAddress(@Query('customerId', ParseIntPipe) customerId: number) {
    return this.addressService.getDefaultAddress(customerId);
  }
}

File: mall-service/system/src/main/java/com/example/system/controller/AddressController.java
package com.example.system.controller;

import com.example.common.result.Result;
import com.example.system.entity.Address;
import com.example.system.service.AddressService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 收货地址控制器
 */
@RestController
@RequestMapping("/address")
@RequiredArgsConstructor
public class AddressController {

    private final AddressService addressService;

    /**
     * 获取用户地址列表
     */
    @GetMapping("/list")
    public Result<List<Address>> getAddressList(@RequestParam Long customerId) {
        List<Address> list = addressService.getAddressList(customerId);
        return Result.success(list);
    }

    /**
     * 获取地址详情
     */
    @GetMapping("/{id}")
    public Result<Address> getAddress(@PathVariable Long id) {
        Address address = addressService.getById(id);
        return Result.success(address);
    }

    /**
     * 添加地址
     */
    @PostMapping("/add")
    public Result<Address> addAddress(@RequestBody Address address) {
        Address result = addressService.addAddress(address);
        return Result.success(result);
    }

    /**
     * 更新地址
     */
    @PutMapping("/update")
    public Result<Void> updateAddress(@RequestBody Address address) {
        boolean success = addressService.updateAddress(address);
        return success ? Result.success() : Result.error("更新失败");
    }

    /**
     * 删除地址
     */
    @DeleteMapping("/{id}")
    public Result<Void> deleteAddress(
            @PathVariable Long id,
            @RequestParam Long customerId) {
        boolean success = addressService.deleteAddress(id, customerId);
        return success ? Result.success() : Result.error("删除失败");
    }

    /**
     * 设置默认地址
     */
    @PutMapping("/default/{id}")
    public Result<Void> setDefault(
            @PathVariable Long id,
            @RequestParam Long customerId) {
        boolean success = addressService.setDefault(id, customerId);
        return success ? Result.success() : Result.error("设置失败");
    }

    /**
     * 获取默认地址
     */
    @GetMapping("/default")
    public Result<Address> getDefaultAddress(@RequestParam Long customerId) {
        Address address = addressService.getDefaultAddress(customerId);
        return Result.success(address);
    }
}
